refactor(curriculum): restructure sudoku solver python project (#54303)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/07-scientific-computing-with-python/learn-classes-and-objects-by-building-a-sudoku-solver/6606cc754a8834509cd0afb6.md

---
id: 6606cc754a8834509cd0afb6
title: Step 36
challengeType: 20
dashedName: step-36
---

# --description--

Another thing to check is if a number can be inserted in a 3x3 square.

> Inside the `Board` class, create a method named `valid_in_square` and give it four parameters: `self`, `row`, `col`, and `num`. Where `row`, `col`, and `num` represent the row index, the column index, and the number to be checked, respectively.

Reviewer: ```suggestion
Inside the `Board` class, define a method named `valid_in_square` with four parameters: `self`, `row`, `col`, and `num`. Here, `row`, `col`, and `num` represent the row index, the column index, and the number to be checked, respectively.
```

# --hints--

You should create a new method named `valid_in_square` within the `Board` class.

```js
({ test: () => assert(runPython(`_Node(_code).find_class("Board").has_function("valid_in_square")`)) })
```

> Your `valid_in_square` method should have four parameters: `self`, `row`, `col`, and `num`.

Reviewer 2: ```suggestion
Your `valid_in_square` method should have four parameters in order: `self`, `row`, `col`, and `num`.
```
